fuse_graphs/src/hash_graph.cpp

```diff
@@ -426,18 +426,18 @@ ceres::Solver::Summary HashGraph::optimize(const ceres::Solver::Options& options
 }
 
 ceres::Solver::Summary HashGraph::optimizeFor(
-  const ros::Duration& max_optimization_time,
+  const std::chrono::nanoseconds& max_optimization_time,
```

Comment:
Good catch! The Graph ABC and its implementations don't depend on ROS in any way, so I was hoping to separate the stack.

That said though, the headers have been migrated, I'm thinking of changing them back to std::chrono :thinking: What do you think?

Reply:
Here: https://github.com/locusrobotics/fuse/pull/283/commits/9314b85ef282a2279c816c771a8d90f591f292d7#diff-d735274778d4bdbb320b2042641a7abe4f36e969e23bcfde0ab87d0f9df53474R382

Reply:
> The Graph ABC and its implementations don't depend on ROS in any way, so I was hoping to keep them separate.
> That said though, the other graph headers have been migrated to rclcpp::Duration, I'm thinking of changing them back to std::chrono thinking What do you think?

It looks like they use `ros::Duration` in ROS 1, so this is actually changing them to `std::chrono`. I'd recommend using ROS types throughout to avoid downstream users having to convert to/from `std::chrono` depending on the API in `fuse_core` they're using.

https://github.com/locusrobotics/fuse/blob/07676a8a47ad00b605153d6d98aa58dce663a39e/fuse_core/include/fuse_core/graph.h#L382

Reply:
https://github.com/locusrobotics/fuse/pull/283/commits/9156f257216c8bed9287df8d23707db59779a674